### PR TITLE
[G7T5-74, G7T5-75] Create FE View for HR to Delete Job Roles

### PR DESCRIPTION
**Pull request by branfats**
# Description
Fixes [Jira ticket 74](https://is212g7t5.atlassian.net/browse/G7T5-74?atlOrigin=eyJpIjoiYzI2N2MyZDU0NzkyNDFmN2FlMDMwNTQwNzY2YjFiMzQiLCJwIjoiaiJ9)

Adds Delete button which Pops up a confirmation. Created softDelete API in jobs.js for ticket 75 (Link BE API to delete Job Roles). Upon "confirm", job role is soft deleted and toastify notification pops up.

## Type of change
Screenshot 2022-10-21 at 1 47 55 PM
___________________________
Screenshot 2022-10-21 at 1 48 25 PM
___________________________
Screenshot 2022-10-21 at 1 48 56 PM

Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?
- [x] I have added the corresponding testcases into the [testcase document](https://docs.google.com/spreadsheets/d/1QOyUN_kFN0fddLkjAQz2DK3jou3cWdN9CJsNbl3v0Kg/edit#gid=0)

Please indicate the testcase ID you have added or are using
- [x] G7T5-3-1
- [ ] G7T5-3-2
- [ ] G7T5-3-3
- [ ] G7T5-3-4
- [ ] G7T5-3-5
- [ ] G7T5-3-6

# Checklist:

- [x] I have added the appropriate labels to my PR
- [x] My code follows the style guidelines of this project
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/src/components/job/hr/JobDeletionPopUp.jsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+          <div className='grid-row py-3 text-lg'>
+            Are you sure you want to delete this job?
+          </div>
+
+          <div className='grid-row py-3 flex justify-end'>
+            <button
+              type='button'
+              className='text-white bg-primary hover:bg-secondary font-medium rounded-lg text-sm px-5 py-2.5 mr-2 mb-2'
+              onClick={() => handleConfirm}
```
Comment (jonathantan1425) [inline app/src/components/job/hr/JobDeletionPopUp.jsx:49]: This does not qualify as an onClick function, it should be `onClick={handleConfirm}`

File: app/src/components/job/hr/JobDeletionPopUp.jsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
+import React, { useState } from "react";
+import { softDelete } from "src/api/jobs";
+import "../JobTile";
```
Comment (jonathantan1425) [inline app/src/components/job/hr/JobDeletionPopUp.jsx:3]: What's this import for? It's introducing a dependency cycle

File: app/src/components/job/hr/JobDeletionPopUp.jsx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+      setErrors(errorList);
+    } else {
+      setErrors([]);
+      setDisplayPopup(true);
+    }
+    //   refreshPage();
+    setTrigger(false);
+  };
+
+  const renderErrors = errors && errors.map && errors.map((error) => <p>{error}</p>);
```
Comment (quinncheong) [inline app/src/components/job/hr/JobDeletionPopUp.jsx:37]: I dont think you need the second errors.map

File: app/src/components/job/hr/JobDeletionPopUp.jsx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  function refreshPage() {
+    window.location.reload(false);
+  }
+
+  const handleConfirm = async (e) => {
+    e.preventDefault();
+    const res = await softDelete(jobId, jobIsActive);
+    if (res.detail) {
+      const errorList = [];
+      if (res.detail.map) {
```
Comment (quinncheong) [inline app/src/components/job/hr/JobDeletionPopUp.jsx:23]: .map is an arr function so doing it will not really check for anything except checking if the method exists --- which it usually does.

File: app/src/components/job/JobTile.jsx [hunk 5]
```diff
@@ -1,22 +1,27 @@
...
+import { updateJob } from "src/api/jobs";
 import SkillBadge from "./SkillBadge";
+import DeletePopUp from "./hr/JobDeletionPopUp";
 
 export default function JobTile({ jobId, jobName, jobDesc, skills, isActive }) {
   const [isDetailsOpen, setIsDetailsOpen] = useState(false);
   const { currentUserType } = useUserContext();
   const { setSelectedJobRole } = useLJCreationContext();
   const { setUpdateJobRole } = useUpdateJobContext();
+  const { setDeleteJobRole } = useUpdateJobContext();
```
Comment (quinncheong) [inline app/src/components/job/JobTile.jsx:22]: Can just put under line 21

File: app/src/components/job/JobTile.jsx [hunk 6]
```diff
@@ -1,22 +1,27 @@
...
   const history = useHistory();
+  const [buttonPopUp, setButtonPopUp] = useState(false);
```
Comment (quinncheong) [inline app/src/components/job/JobTile.jsx:24]: Name it more intuititve like 'isButtonPopupOpen"

**Review by quinncheong — COMMENTED** [review]
Left some general comments. Hopefully it helps

File: app/src/components/job/hr/JobDeletionPopUp.jsx (hunk 4 above)
Comment (branfats) [reply app/src/components/job/hr/JobDeletionPopUp.jsx:23]: Ok will remove in upcoming commit!

**Review by jonathantan1425 — COMMENTED** [review]
Not sure if using a whole new package here is the best idea. Either we'll have to switch out all other pop up variations to this, or the other way round

File: app/src/routes/deleteJob/index.jsx [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+
+//   return (
+//     <div className="w-11/12 max-w-7xl mx-auto">
+//       <LJDeletionPopUp/>
+//     </div>
+//   )
+// }
+
+
+// To be deleted
```
[the remark this answers is not in the document]
Comment (branfats) [reply app/src/routes/deleteJob/index.jsx:13]: oops my bad forgot to delete this. Will do so now

File: app/src/components/job/JobTile.jsx [hunk 8]
```diff
@@ -1,22 +1,26 @@
...
+import { updateJob } from "src/api/jobs";
 import SkillBadge from "./SkillBadge";
+import DeletePopUp from "./hr/JobDeletionPopUp";
 
 export default function JobTile({ jobId, jobName, jobDesc, skills, isActive }) {
   const [isDetailsOpen, setIsDetailsOpen] = useState(false);
   const { currentUserType } = useUserContext();
   const { setSelectedJobRole } = useLJCreationContext();
-  const { setUpdateJobRole } = useUpdateJobContext();
+  const { setUpdateJobRole, setDeleteJobRole } = useUpdateJobContext();
```
[the remark this answers is not in the document]
Comment (branfats) [reply app/src/components/job/JobTile.jsx:21]: Don't think so. Think this was from Update Job Role ticket, so I didn't touch this

[the remark this answers is not in the document]
Comment (branfats) [reply app/src/components/job/JobTile.jsx:21]: I might have accidentally added it but I have no recollection for real. I'll delete it since it's not used at all

**Comment by branfats** [comment]
Understand the concern about the use of the new package. 

I used it initially cause it's a pop up after a pop-up and I felt that it was difficult rendering a second pop up after the first one (first pop up layer overriding the other). When trigger was set to false for the first pop up to close it, the second pop up wouldn't show. Attempted to move the success pop up into Job Tile where it is a sibling to delete pop up but had trouble coding it out, so I decided to use this new package. Other components didn't have double pop-ups so I was quite lost. Spent a lot of time trying to code it and wasn't able to find any solutions online.

What do the others think? Is it ok to implement the use of sweetalert package for this case?

**Comment by quinncheong** [comment]
I'm not too keen on having a new popup but I guess if it does the job its fine. Is it a pop up after a popp? Sorry the images don't really show that it is atm. Also @deadline2359 did something similar with a double popup system so maybe you can check out her code / ask her.

TLDR merge if you think its ok and fulfils the AC.

**Comment by branfats** [comment]
@quinncheong Apologies, updated the screenshots in the ticket description already.
Fulfils acceptance criteria except for this which I created a new subtask for
Staff with Learning Journeys (LJ) created corresponding to these Job Roles:
-Can still view their LJ and its details
-Can no longer update LJ (via addition/deletion of courses)
-Job Role will be indicated to be Inactive for the LJ

**Comment by quinncheong** [comment]
Hmm we dont need this popup then actually. Because the second popup (actually a modal), is just to provide the user with information right? 

You can use a 'toast' instead to achieve the same outcome, and we already have a toast inside the codebase. Maybe you can just swap to the toast and that should do the trick.

**Comment by branfats** [comment]
Ok, I thought had to use popup cause a popup or Modal was used for UpdateJobs, not toast. But ok I will try to change to toast tonight

**Review by quinncheong — APPROVED** [review]
